### PR TITLE
Adaptive reordering fixes

### PR DESCRIPTION
**Pull request by ghedo**
Some additional fixes to adaptive reordering thresholds.

### CODE REVIEW

File: src/recovery/cubic.rs [hunk 1]
```diff
@@ -213,13 +213,12 @@ fn on_packet_acked(
...
-        if let Some(prior) = &r.cubic_state.prior {
-            let new_lost = r.lost_count - prior.lost_count;
+        let new_lost = r.lost_count - r.cubic_state.prior.lost_count;
 
-            if r.congestion_window < prior.congestion_window &&
-                new_lost < RESTORE_COUNT_THRESHOLD
-            {
-                rollback(r);
+        if new_lost < RESTORE_COUNT_THRESHOLD {
+            let did_rollback = maybe_rollback(r);
```
Comment (junhochoi) [inline src/recovery/cubic.rs:219]: What's the reason behind this change? Here we can safely assume checkpoint() is already called (it's already in recovery episode), calling rollback() doesn't need additional check.

Comment (ghedo) [reply src/recovery/cubic.rs:219]: See the second commit's message. This is just to make the rollback avoidance if prior cwnd is smaller more generic, so each called to `rollback()` doesn't need to check this (i.e. we need this for adaptive reordering too).

Comment (ghedo) [reply src/recovery/cubic.rs:219]: So basically this moves the ``if r.congestion_window < prior.congestion_window` into `rollback()` directly, but here we need to check if the rollback actually happened (or at least that's what the previous code did), hence `maybe_rollback()`.

Comment (junhochoi) [reply src/recovery/cubic.rs:219]: I see. If so, why not adding `bool` type to `rollback()` and merge with `maybe_rollback()`?

Comment (ghedo) [reply src/recovery/cubic.rs]: No real reason. I just thought this might be slightly smaller, but your way is fine to me too. Updated commit to do that.

File: src/recovery/cubic.rs [hunk 2]
```diff
@@ -213,13 +213,12 @@ fn on_packet_acked(
...
 
-            if r.congestion_window < prior.congestion_window &&
-                new_lost < RESTORE_COUNT_THRESHOLD
-            {
-                rollback(r);
+        if new_lost < RESTORE_COUNT_THRESHOLD {
+            let did_rollback = rollback(r);
+
+            if did_rollback {
                 return;
```
Comment (junhochoi) [inline src/recovery/cubic.rs:222]:
```suggestion
            if rollback(r) {
                return;
```

**Review by junhochoi — APPROVED** [review]

File: src/recovery/cubic.rs (hunk 2 above)
Comment (ghedo) [reply src/recovery/cubic.rs:222]: So, I did that originally, but then clippy complained that the `if`s could be merged together, though I found:
```rust
if new_lost < RESTORE_COUNT_THRESHOLD && rollback() {
```
kind of hard to read (as the `rollback()` call looks a bit hidden, and would be easy to pass over when reading), so I did it this way to silence clippy.

Comment (junhochoi) [reply src/recovery/cubic.rs:222]: no problem then. 👍
